vector/src/main/java/im/vector/app/features/autocomplete/member/AutocompleteMemberPresenter.kt

```diff
@@ -127,9 +125,14 @@ class AutocompleteMemberPresenter @AssistedInject constructor(
                     context.getString(R.string.room_message_autocomplete_users)
             )
 
-    private fun createMemberItems(queryParams: RoomMemberQueryParams) =
+    private fun createMemberItems(queryParams: RoomMemberQueryParams, query: CharSequence?) =
             room.membershipService()
                     .getRoomMembers(queryParams)
+                    .filter { roomMemberSummary ->
+                        query?.let {
+                            it.isBlank() || roomMemberSummary.displayName?.contains(it, true).orFalse() || roomMemberSummary.userId.contains(it, true)
+                        }.orTrue()
+                    }
```

Comment:
I'm wondering if it's better to add the filter on the resulting list or in the realm query:

```
    override fun getRoomMembers(queryParams: RoomMemberQueryParams, query: String): List<RoomMemberSummary> {
        return monarchy.fetchAllMappedSync(
                {
                    roomMembersQuery(it, queryParams)
                            .and()
                            .contains(RoomMemberSummaryEntityFields.USER_ID, query)
                            .or()
                            .contains(RoomMemberSummaryEntityFields.DISPLAY_NAME, query)
                },
                {
                    it.asDomain()
                }
        )
    }
```

WDYT?

Reply:
Maybe add `val displayNameOrUserId: QueryStringValue,` to the data class `RoomMemberQueryParams`?